src/FilterGroup.php

```diff
@@ -19,6 +22,20 @@ public function withFilter(Filter $filter): self
         return $this;
     }
 
+    public function withLongFilter(Filter $filter, int $offset = 0, int $lenght = 200): self
+    {
+        $this->longFilters[] = [
+            'field' => $filter->field,
+            'value' => $filter->value,
+            'condition_type' => $filter->conditionType,
+        ];
+
+        $this->offset = $offset;
+        $this->lenght = $lenght;
+
+        return $this;
+    }
```

Comment:
J'ai un problème avec cette implémentation. La valeur de `$this->offset` et `$this->lenght` sera celle du dernier appel à `withLongFilter()`. C'est un couplage temporel qui aura des conséquences difficiles à prévoir.